torch/optim/adamw.py

```diff
@@ -25,189 +23,13 @@ def __init__(
-                    )
-
-            exp_avgs.append(state["exp_avg"])
-            exp_avg_sqs.append(state["exp_avg_sq"])
-
-            if group['amsgrad']:
-                max_exp_avg_sqs.append(state["max_exp_avg_sq"])
-            if group['differentiable'] and state['step'].requires_grad:
-                raise RuntimeError('`requires_grad` is not supported for `step` in differentiable mode')
-
-            # Foreach without capturable does not support a tensor lr
-            if group['foreach'] and isinstance(group['lr'], Tensor) and not group['capturable']:
-                raise RuntimeError('lr as a Tensor is not supported for capturable=False and foreach=True')
-
-            state_steps.append(state["step"])
-        return has_complex
+        super().__init__(params, lr, betas, eps, weight_decay, amsgrad, foreach=foreach,
+                         maximize=maximize, capturable=capturable, differentiable=differentiable,
+                         fused=fused, decoupled_weight_decay=True)
 
     @_use_grad_for_differentiable
     def step(self, closure=None):
-        """Perform a single optimization step.
-
-        Args:
-            closure (Callable, optional): A closure that reevaluates the model
-                and returns the loss.
-        """
-        self._cuda_graph_capture_health_check()
-
-        loss = None
-        if closure is not None:
-            with torch.enable_grad():
-                loss = closure()
-
-        for group in self.param_groups:
-            params_with_grad = []
-            grads = []
-            exp_avgs = []
-            exp_avg_sqs = []
-            max_exp_avg_sqs = []
-            state_steps = []
-            amsgrad = group["amsgrad"]
-            beta1, beta2 = group["betas"]
-
-            has_complex = self._init_group(
-                group,
-                params_with_grad,
-                grads,
-                amsgrad,
-                exp_avgs,
-                exp_avg_sqs,
-                max_exp_avg_sqs,
-                state_steps,
-            )
-
-            adamw(
-                params_with_grad,
-                grads,
-                exp_avgs,
-                exp_avg_sqs,
-                max_exp_avg_sqs,
-                state_steps,
-                amsgrad=amsgrad,
-                beta1=beta1,
-                beta2=beta2,
-                lr=group["lr"],
-                weight_decay=group["weight_decay"],
-                eps=group["eps"],
-                maximize=group["maximize"],
-                foreach=group["foreach"],
-                capturable=group["capturable"],
-                differentiable=group["differentiable"],
-                fused=group["fused"],
-                grad_scale=getattr(self, "grad_scale", None),
-                found_inf=getattr(self, "found_inf", None),
-                has_complex=has_complex,
-            )
-
-        return loss
+        super().step(closure)
```

Comment:
@mlazos it is looking like this is causing a graph break in the dynamo tests--do you have pointers?

Reply:
```super().step()``` has been decorated by ```_use_grad_for_differentiable``` which introduces a graph break, so here you apply the decorator twice. Probably we just need to remove L30.

Reply:
Ooooh thank you!

Reply:
Hi @yanboliang , I went ahead and removed L30 but it looks like another error is being thrown in test_optimizers.py, specifically in the unwrap step. Since we've removed the wrapper on adamw should we add some logic to access opt.step._wrapped_ strictly in the case it exists, or is there a better approach to take here?

Reply:
Yea, I think you can add a check ```if hasattr(opt.step, "__wrapped__"):```.

Reply:
I think we can be bigger brain and remove this redefinition of step entirely! it will just call Adam.step anyway :p 

Then, the hacks added in the test files no longer need to be there.